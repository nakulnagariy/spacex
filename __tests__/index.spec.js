import * as React from 'react'
import { mount, shallow } from 'enzyme'
import SpacexPage from '../pages/index'
import { data } from '../mock'

beforeAll(() => {
  global.fetch = jest.fn()
  //window.fetch = jest.fn(); if running browser environment
})

// src/utils/currency.test.js
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(data),
  })
)

describe('Pages', () => {
  describe('SpaceX', () => {
    it('should render without throwing an error', () => {
      const wrap = mount(<SpacexPage missions={data} />)
      expect(wrap).toMatchSnapshot()
      expect(wrap.find('button').length).toBe(11)
      expect(wrap.find({ className: 'card' }).length).toBe(10)
    })

    it('Filter the result', async () => {
      const wrap = mount(<SpacexPage missions={data} />)
      wrap.find('button').last().simulate('click')
      wrap.update()
      expect(wrap.props().missions.length).toBe(10)
    })
  })
})
