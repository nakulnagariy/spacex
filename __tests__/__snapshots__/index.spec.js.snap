// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages SpaceX should render without throwing an error 1`] = `ReactWrapper {}`;
